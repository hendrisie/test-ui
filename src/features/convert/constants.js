export const NAME = 'convert';

export const ACTION_SUBMIT = 'features/convert/ACTION_SUBMIT';
