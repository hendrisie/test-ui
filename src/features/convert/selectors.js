import { createSelector } from 'reselect';
import { NAME } from './constants';

const getResultState = state => {
  if (!(state && state[NAME] && state[NAME].result)) return null;
  return state[NAME].result;
};

const getResult = createSelector(
  [getResultState],
  result => result
);

const getLeftoverState = state => {
  if (!(state && state[NAME] && state[NAME].leftover)) return null;
  return state[NAME].leftover;
};

const getLeftover = createSelector(
  [getLeftoverState],
  leftover => leftover
);

export default {
  getResult,
  getLeftover,
};
