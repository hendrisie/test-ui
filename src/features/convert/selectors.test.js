import reducer, { initialState } from './reducers';
import { NAME, ACTION_SUBMIT } from './constants';
import selectors from './selectors';

describe('selectors', () => {
  let state;
  beforeEach(() => {
    state = {
      [NAME]: reducer(initialState, {
        type: ACTION_SUBMIT,
        payload: 'Rp 100501'
      })
    };
  });
  it('should return correct getResult', () => {
    expect(selectors.getResult(state)).toMatchSnapshot();
  });

  it('should return correct getLeftover', () => {
    expect(selectors.getLeftover(state)).toMatchSnapshot();
  });
});
