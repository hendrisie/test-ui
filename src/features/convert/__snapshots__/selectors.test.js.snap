// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors should return correct getLeftover 1`] = `1`;

exports[`selectors should return correct getResult 1`] = `
Array [
  Array [
    1,
    100000,
  ],
  Array [
    1,
    500,
  ],
]
`;
